lib/src/client/dio_interceptor.dart

```diff
@@ -0,0 +1,226 @@
+
+      // Build parameters for constructing the signature body
+      final signatureBodySourceParameters = SignatureBodySourceParameters(
+          options.method,
+          options.uri,
+          options.headers
+              .map((key, value) => MapEntry(key.toString(), value.toString())),
+          _uuid.v4(), // Generate a nonce
+          _getTimestamp(options.headers[HttpHeaders.dateHeader]),
+          _options.clientId!,
+          signatureBodySourceComponents,
+          body: body);
+
+      // Build the signature body source based on parameters
+      final signatureBodySource = await _signatureBodySourceBuilder
+          .build(signatureBodySourceParameters);
+
+      // Construct parameters for creating the signature
+      final signatureBodyParameters =
+          SignatureBodyParameters(signatureBodySource, _options.clientSecret!);
+
+      // Generate the signature using the signer
+      final signatureBody =
+          await _signatureBodySigner.sign(signatureBodyParameters);
+
+      // Format the signature header based on the specified pattern
+      final signatureHeader = _options.signaturePattern
+          .replaceAll("{ClientId}", _options.clientId!)
+          .replaceAll("{Nonce}", signatureBodySourceParameters.nonce)
+          .replaceAll(
+              "{Timestamp}", signatureBodySourceParameters.timestamp.toString())
+          .replaceAll("{SignatureBody}", signatureBody);
+
+      print('[DEBUG] Generated Signature Header: $signatureHeader');
+
+      // Add the signature header to the request headers
+      options.headers[_options.headerName] = signatureHeader;
+    } catch (error, stackTrace) {
+      print('[x] _signRequest error');
+      throw RequestsSignatureException('$error\n$stackTrace');
+    }
+  }
+
+  // Returns what the client thinks is the current time.
+  static int getTime(DateTime? timestamp) {
+    if (timestamp != null && timestamp.toString().isNotEmpty) {
+      print('[i] _getTime called with timestamp: $timestamp');
+
+      return timestamp.toUtc().millisecondsSinceEpoch ~/ 1000;
+    } else {
+      return DateTime.now().toUtc().millisecondsSinceEpoch ~/ 1000;
+    }
+  }
+
+  // Returns the timestamp, accounting for the perceived clock skew.
+  int _getTimestamp(DateTime? timestamp) {
+    print('[i] clockSkew: $_clockSkew');
+    return getTime(timestamp) + _clockSkew;
+  }
+
+  // Parses unexpected date strings formats into a DateTime object.
+  DateTime parseDateString(String dateString) {
+    List<String> possibleFormats = [
+      'E, d MMM yyyy HH:mm:ss zzz', // Standard HttpsHeaders.dateHeader format
+      'EEE, dd MMM yyyy HH:mm:ss zzz', // Alternative format with full day name
+      'EEEE, dd MMMM yyyy HH:mm:ss zzz' // Full day and month names format
+    ];
+
+    for (String formatString in possibleFormats) {
+      try {
+        DateFormat format = DateFormat(formatString);
+        return format.parse(dateString);
+      } catch (e) {
```

Comment:
Relying on whether the parse throws exceptions or not it probably not the right way of try-parsing. Throwing exceptions usually have a performance cost because exception capture the stack trace. I suspect there is a way to avoid the exceptions. There's probably a standard way of parsing that date format.